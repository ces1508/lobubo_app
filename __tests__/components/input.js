import React from 'react'
import Input from '../../src/components/input'
import Icons from 'react-native-vector-icons/MaterialCommunityIcons'

import rendered from 'react-test-renderer'

test('render correctly', () => {
  const tree = rendered.create(<Input
    onChangeText= {text => text}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render with leftComponent', () => {
  const tree  = rendered.create(<Input
    onChangeText={text => text}
    leftIcon={<Icons name='account' />}
    />)
    expect(tree).toMatchSnapshot()
})