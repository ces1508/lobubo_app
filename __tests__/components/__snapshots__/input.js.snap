// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .3)",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 10,
      "marginVertical": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "white",
          "flex": 1,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`render with leftComponent 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .3)",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 10,
      "marginVertical": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï€„
  </Text>
  <TextInput
    allowFontScaling={true}
    leftIcon={
      <Icon
        allowFontScaling={false}
        name="account"
        size={12}
      />
    }
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "white",
          "flex": 1,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
